src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java

```diff
@@ -600,29 +604,56 @@ public Response showSettings(@Context HttpServletRequest request) throws IOExcep
 	@POST
 	@Path("/branches/delete")
 	@Transactional
-	public Response deleteBehindBranch(@Context HttpServletRequest request,
-									   @FormParam("branchDeleteName") String branchDeleteName,
-									   @QueryParam("fatal") String fatal)
+	public Response deleteBranch(@Context HttpServletRequest request,
+                                 @FormParam("branchDeleteName") String branchDeleteName,
+                                 @FormParam("branchDeleteNameConfirmation")
+                                             String branchDeleteNameConfirmation,
+                                 @QueryParam("fatal") String fatal)
 			throws IOException, ApiError {
 
 		RepositoryEntity repositoryEntity = getRepositoryEntity();
 		RepositoryApi repositoryApi = repositoriesApi.getRepository(repositoryEntity.getRepositoryName());
 
         List<Locale> locales = Collections.list(request.getLocales());
 
-        Map<String, Object> parameters = getBaseParameters();
+        Map<String, Object> parameters;
 
         if (branchDeleteName != null) {
             BranchApi branchApi = repositoryApi.getBranch(branchDeleteName);
 			BranchModel branchModel = branchApi.get();
 
 			if (!branchModel.isAhead()) {
+                // Branch is not ahead and can be safely removed
 				branchApi.deleteBranch();
-				parameters = getBranchOverviewParameters("master", 1, true);
+				parameters = getBranchOverviewParameters("master", 1, "success");
 			} else {
-				parameters = getBranchOverviewParameters("master", 1, false);
+                // Branch is ahead
+                if (branchDeleteNameConfirmation != null) {
+					String branchSimpleName = branchDeleteName.split("refs/heads/")[1];
```

Comment:
It's a bit smelly yeah, but the problem is, we want the confirmation the user types in to be the simple name (not refs/heads etc.). However if I check if the branchName ends with the confirmation, then typing in the last character of the branchName will succesfully delete it.

I could maybe split it up at every slash, and then get the last item in the array, which should be the branch name. This will however be prone to errors on branches including slashes. Hmm.